src/components/data/DataTable.test.ts

```diff
@@ -0,0 +1,424 @@
+import { DeepPartial } from '@terra-ui-packages/core-utils';
+import { act, fireEvent, render, screen, within } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import _ from 'lodash/fp';
+import { useState } from 'react';
+import { h } from 'react-hyperscript-helpers';
+import { defaultGoogleWorkspace } from 'src/analysis/_testData/testData';
+import DataTable from 'src/components/data/DataTable';
+import { Ajax } from 'src/libs/ajax';
+import { DataTableProvider } from 'src/libs/ajax/data-table-providers/DataTableProvider';
+import { asMockedFn } from 'src/testing/test-utils';
+
+type AjaxContract = ReturnType<typeof Ajax>;
+
+jest.mock('src/libs/ajax');
+
+type ReactNotificationsComponentExports = typeof import('react-notifications-component');
+jest.mock('react-notifications-component', (): DeepPartial<ReactNotificationsComponentExports> => {
+  return {
+    Store: {
+      addNotification: jest.fn(),
+      removeNotification: jest.fn(),
+    },
+  };
+});
+
+const entities = _.map(
+  (n) => ({ entityType: 'sample', name: `sample_${n}`, attributes: { attr: n % 2 === 0 ? 'even' : 'odd' } }),
+  _.range(0, 250)
+);
+
+type ReactVirtualizedExports = typeof import('react-virtualized');
+jest.mock('react-virtualized', (): ReactVirtualizedExports => {
+  const actual = jest.requireActual<ReactVirtualizedExports>('react-virtualized');
+
+  const { AutoSizer } = actual;
+  class MockAutoSizer extends AutoSizer {
+    state = {
+      height: 7000,
+      width: 2000,
+    };
+
+    setState = () => {};
+  }
+
+  return {
+    ...actual,
+    AutoSizer: MockAutoSizer,
+  };
+});
+
+const getPage = jest
+  .fn()
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  .mockImplementation((signal, entityType, queryOptions: { pageNumber; columnFilter }, entityMetadata) => {
```

Comment:
Consider adding brief comment explaining the intended behavior of the mock?  (eg: pretends to be X pages with Y items)